presto-cassandra/src/main/java/com/facebook/presto/cassandra/NativeCassandraSession.java

```diff
@@ -334,7 +336,16 @@ private CassandraColumnHandle buildColumnHandle(TableMetadata tableMetadata, Col
     @Override
     public List<CassandraPartition> getPartitions(CassandraTable table, List<Set<Object>> filterPrefixes)
     {
-        Iterable<Row> rows = queryPartitionKeys(table, filterPrefixes);
+        Iterable<Row> rows;
+        if (getCassandraVersion().compareTo(PARTITION_FETCH_OPTIMIZE_VERSION) > 0) {
+            log.debug("Using optimized partition fetcher.");
```

Comment:
This is cryptic, nobody would know what it means. Better would be to say that you are using `IN` predicate to select partitions.